app/abilities/memberships/terminate_sac_membership_ability.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2012-2024, Schweizer Alpen-Club. This file is part of
+#  hitobito_sac_cas and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito_sac_cas.
+
+module Memberships
+  class TerminateSacMembershipAbility < AbilityDsl::Base
+    on(Wizards::Memberships::TerminateSacMembershipWizard) do
+      permission(:any).may(:create).for_self_if_active_member_or_backoffice
+    end
+
+    def for_self_if_active_member_or_backoffice
+      active_member? && (for_self? || backoffice?)
+    end
+
+    def backoffice?
+      user_context.user.backoffice?
+    end
+
+    def for_self?
+      subject.person == user_context.user
+    end
```

Comment:
The permissions are very similar for all these wizard permissions. We could think about DRYing this up in the future (but I wouldn't do it in this PR)

Reply:
I think this is a good idea